### ISSUE TITLE
MediaPipe Web Roadmap

### ISSUE THREAD

**Issue by mheskol** [google/mediapipe#2187]
**System information** (Please provide as much relevant information as possible)

- MediaPipe Solution (you are using): Object Detection
- Programming language : Javascript
- Are you willing to contribute it: Yes:

Is there a highlevel roadmap you can share about plans for MediaPipe. Especially expanding the support of MediaPipe Web to cover other modules available only on other platforms like Object Detection and KNIFT

I think that will be useful to the community overall. Of course the more information you know and can share the better but even if not much is known and it is still rough plans it is can still be useful.

**Comment by sgowroji** [comment]
Hi @mheskol, Could you please elaborate your feature request on Object detection.  Which use case are you looking in web. Thanks!

**Comment by mheskol** [comment]
> 
> 
> Hi @mheskol, Could you please elaborate your feature request on Object detection. Which use case are you looking in web. Thanks!

Hi @sgowroji and thanks for your response
Actually we are using Object Detection already on mobile and looking to provide similar functionality to our users on Web.
We use it to detect cards of various types (business cards, credit cards, driving license) or documents in near realtime to assist the user in capturing good image of the object (focus on object, positionining, ....etc.)

The Object Detection module available on Android and iOS is working well for us and it would be very useful to have it on Web as well.
If porting it to Web is not in your short to med-term plans yet, it would be appreciated if you are able to give an idea of what the effort involved and the appraoch to port it as we can consider contributing to this as well.

Thanks

**Comment by mheskol** [comment]
Hi @mgyong . It seems my question has been assigned to you. Have you had the chance to look at it and do you have an update you can share at this stage please?
It would be very helpful if you can share what you know currently as we are invested in using MediaPipe and we want to feed into our med-term planning process
Thanks

**Comment by mgyong** [comment]
@mheskol Sorry for late reply. We currently don't have plans for porting object detection to the web. That said, we are trying to open source some version of  the MediaPipe web framework so that developers can use their MP graphs on the web. Its interesting that your company has object detection for android & iOS running. Would be great if you could share your company name and use case, we would potentially be interested to have a blog post on this to share with other MP developers

**Comment by tyrmullen** [comment]
We did release the MediaPipe Objectron JS Solution API for object detection, although I'm not sure if that fits into your use case or not.  See [codepen demo](https://codepen.io/mediapipe/pen/BaWvzdY) and [reference documentation](https://google.github.io/mediapipe/solutions/objectron.html#javascript-solution-api).

Also, there has been some progress in open-sourcing the WebAssembly portion of the MediaPipe web framework, although it is partial/experimental, so I'd still recommend using one of our Solutions if possible.  You can track the latest updates in issue #877 .

**Comment by mheskol** [comment]
> @mheskol Sorry for late reply. We currently don't have plans for porting object detection to the web. That said, we are trying to open source some version of the MediaPipe web framework so that developers can use their MP graphs on the web. Its interesting that your company has object detection for android & iOS running. Would be great if you could share your company name and use case, we would potentially be interested to have a blog post on this to share with other MP developers

Thanks for the response @mgyong . My company name is IDWise (idwise.com) and we used object detection as part of our smart document processing solution. It allows businesses to add smart document capture capabilities to their apps and use the platform to extract data using AI, NLP, ...etc.

Yes open sourcing these frameworks would be very brilliant idea and would allow the community to contribute in ways that Google/MP teams can't prioritise for now for the benefit of everybody. Is open-sourcing something you will be able to do soon?

@tyrmullen Thanks for pointing out these suggestions. We did experiment with the Objectron JS solution but it doesn't fit our use-case but the Object Detector was a much better fit. I will look into #877 as well but will really appreicate any resources that could be open sourced towards more capabilities on the web.

Thanks
